server/course/service.go

```diff
@@ -96,7 +101,14 @@ func CreateCourse(ctx context.Context, course courseDTO.CreateCourse) (courseDTO
 	// create keycloak roles
 	err = CourseServiceSingleton.createCourseGroupsAndRoles(ctx, createdCourse.Name, createdCourse.SemesterTag.String)
 	if err != nil {
-		log.Debug("Failed to create keycloak roles for course: ", err)
+		log.Error("Failed to create keycloak roles for course: ", err)
+		return courseDTO.Course{}, err
+	}
+
+	roleString := fmt.Sprintf("%s-%s-Lecturer", createdCourse.Name, createdCourse.SemesterTag.String)
+	err = CourseServiceSingleton.addUserToGroup(ctx, requesterID, roleString)
```

Comment:
What happens to the database object when this call fails? It should be rolled back, otherwise we have a database entry without its respective roles

Reply:
it is rolled back, as it happens within a transaction. If an error occurs, the function exits early and the transaction is never committed.

Reply:
I highly doubt that this applies to actions which are executed on the Keycloak Groups and roles. Which my comment was actually referring to